### PR TITLE
preflight: detect gateway port 18789 conflicts during onboard

### LINKED ISSUE #18: Detect host `openclaw-gateway.service` port conflicts and fail loudly before onboarding continues
## Summary
If a host-level OpenClaw gateway is already running, it can steal port `18789` and cause NemoClaw onboarding to continue with a dead dashboard.

## Problem
The current flow prints an error but still proceeds, leaving users with a partially configured system and no clear next step.

## Expected behavior
Onboarding should detect the conflict, identify the process or service holding the port, and stop before claiming success.

## Acceptance criteria
- Onboarding checks whether the required gateway port is already in use.
- The error identifies the conflicting process or service when possible.
- Onboarding exits non-zero instead of continuing silently.
- The troubleshooting docs include the systemd user-service case and the recovery workflow.

### LINKED ISSUE #51: ./install.sh assumes port 8080 isn't already in use
I have other Docker containers running on my system; of which, one is already using 8080.  I think your script should validate the ports it wants to use are not already in use before attempting to install.

### PR DESCRIPTION
**Pull request by brianwtaylor**
## Summary
- Add `checkPortAvailable()` to `bin/lib/preflight.js` — uses `lsof` (primary) to identify blocking process name + PID, with a Node.js `net` probe fallback for cross-platform `EADDRINUSE` detection
- Check both required ports: **8080** (OpenShell gateway) and **18789** (NemoClaw dashboard)
- Integrate into onboard step 1 (after cgroup check, before GPU detection) so users get a clear error with remediation steps instead of a silent failure
- 9 test cases covering lsof parsing, net probe, multi-port, and smoke tests

Closes #18 — detect host `openclaw-gateway.service` port conflicts and fail loudly before onboarding continues
Addresses #51 — `install.sh` assumes port 8080 isn't already in use

## Problem
`nemoclaw onboard` starts the gateway on port 8080 and the dashboard on port 18789. If something else (e.g., a leftover `openclaw-gateway.service`, another container, or a dev server) is already listening, the gateway bind fails silently and the user ends up with a non-functional dashboard.

## Detection chain
1. **lsof** (primary): `lsof -i :PORT -sTCP:LISTEN -P -n` — works on Linux + macOS, returns process name + PID. Uses `-sTCP:LISTEN` to avoid false positives from TIME_WAIT connections.
2. **Node.js net probe** (fallback): `net.createServer().listen(port)` — cross-platform, detects EADDRINUSE but can't identify the process. Catches cases where non-root users can't see root-owned listeners via lsof.
3. If `lsof` is not installed, falls through to the net probe automatically

### Automated Tests
```
node --test test/preflight.test.js
```

- 9 tests: lsof parsing (single + multi listener), net probe (free + occupied), empty lsof fallthrough, smoke test, default port, gateway port 8080

### Manual verification
```bash
# Start a dummy server on 18789
node -e "require('net').createServer().listen(18789)" &
# Run onboard — should fail fast with clear error
nemoclaw onboard
# Kill dummy server
kill %1

# Same for gateway port
node -e "require('net').createServer().listen(8080)" &
nemoclaw onboard
# Should report port 8080 conflict with process info
kill %1
```

Closes #51
Closes #18

### CODE REVIEW

File: bin/lib/preflight.js [hunk 1]
```diff
@@ -77,4 +78,85 @@ function checkCgroupConfig(opts) {
...
+    }
+
+    if (typeof lsofOut === "string") {
+      // lsof ran — empty output means port is free
+      const lines = lsofOut.split("\n").filter((l) => l.trim());
+      // Skip the header line (starts with COMMAND)
+      const dataLines = lines.filter((l) => !l.startsWith("COMMAND"));
+      if (dataLines.length === 0) {
+        return { ok: true };
+      }
```
[the remark this answers is not in the document]
Comment (brianwtaylor) [reply bin/lib/preflight.js]: You're right — I was wrong here. Investigated further and confirmed empirically on the target hardware:

Non-root `lsof` cannot see root-owned listeners. It reads `/proc/[pid]/fd` which returns `EACCES` for root-owned PIDs, so it silently drops the socket from output. The kernel's `bind()` has no such limitation — it checks the socket hash table directly.

This is the exact NemoClaw deployment scenario: Docker proxy or a leftover gateway runs as root, non-root user runs `nemoclaw onboard`, lsof returns empty, and we'd incorrectly declare the port free.

Fixed in 8e18771 — empty lsof output now falls through to the net probe. Also added a test that proves the scenario: injected empty lsof + occupied port = net probe catches EADDRINUSE.

**Review by ericksoa — APPROVED** [review]
LGTM — solid architecture with lsof + net probe fallback, good test coverage. Tested: port conflict correctly detected and blocks onboard, clean ports pass preflight.
